decidim-participatory_processes/app/services/decidim/participatory_processes/automate_processes_steps.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ParticipatoryProcesses
+    # This service change the processes steps automatically.
+    class AutomateProcessesSteps
+      def initialize
+        @participatory_processes = Decidim::ParticipatoryProcess.published.where("start_date <= ? AND end_date >= ?", Time.zone.now.to_date, Time.zone.now.to_date)
+      end
+
+      def change_active_step
```

Comment:
Can we move the `@participatory_processes` query into the `change_active_step` method and get rid of the initializer please?